src/configurationProvider.ts

```diff
@@ -153,14 +151,26 @@ export class CSharpConfigurationProvider implements vscode.DebugConfigurationPro
 	 */
     resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, config: vscode.DebugConfiguration, token?: vscode.CancellationToken): vscode.ProviderResult<vscode.DebugConfiguration> {
 
-        if (!config.type)
-        {
+        if (!config.type) {
             // If the config doesn't look functional force VSCode to open a configuration file https://github.com/Microsoft/vscode/issues/54213
             return null;
         }
 
-        if (config.request === "launch")
-        {
+        if (config.request === "attach" && config.processCommand) {
+            const processCommand = config.processCommand.replace(/\${workspaceFolder}/g, folder.uri.fsPath);
+            delete config.processCommand;
+            return this.attachItemsProvider.getAttachItems()
+                .then(attachItems => {
+                    const attachItem = attachItems.find(ai => ai.detail === processCommand);
+                    if (attachItem == null) {
+                        throw new Error("Can't find process");
+                    }
+                    config.processId = attachItem.id;
```

Comment:
>config.processId = attachItem.id; [](start = 20, length = 33)

We should probably fail if there are multiple processes with the command rather than picking the first.

Reply:
Done